### PR TITLE
param: upgraded json_tok_number, ...

### PR DESCRIPTION
**Pull request by wythe**
`json_tok_u64, json_tok_double, and json_tok_bool`.

Also renamed old versions for number, u64, and double to:
-   `json_to_number`
-   `json_to_u64`
-   `json_to_double`

for use as utility functions.

This has turned into a **WIP**. It's going well. I can continue until completely converted. I can even fold in #1823. Whatever would be easiest for reviewers.

### CODE REVIEW

File: lightningd/peer_control.c [hunk 1]
```diff
@@ -1169,22 +1169,15 @@ static void json_dev_forget_channel(struct command *cmd, const char *buffer,
...
 	forget->cmd = cmd;
 
-	/* If &forget->force is used directly in p_opt_def() below then
-	 * gcc 7.3.0 fails with:
-	 * 'operation on ‘forget->force’ may be undefined [-Werror=sequence-point]'
-	 *
-	 * See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86584
-	 *
-	 * Hence this indirection.
-	 */
```
Comment (wythe) [inline lightningd/peer_control.c]: The indirection is now necessary because the callback now requires an "address of pointer".  So I removed the comment.

Comment (rustyrussell) [reply lightningd/peer_control.c]: Yep, great.

**Comment by wythe** [comment]
Made good progress today.  Almost 100% converted.

**Comment by wythe** [comment]
Victory, I think.

File: lightningd/json.c [hunk 2]
```diff
@@ -104,12 +201,26 @@ void json_add_short_channel_id(struct json_result *response,
...
+}
+
+bool json_tok_short_channel_id(struct command *cmd, const char *name,
+			       const char *buffer, const jsmntok_t *tok,
+			       struct short_channel_id **scid)
+{
+	*scid = tal(cmd, struct short_channel_id);
+	if (short_channel_id_from_str(buffer + tok->start,
+				      tok->end - tok->start, *scid))
+		return true;
```
[the remark this answers is not in the document]
Comment (wythe) [reply lightningd/json.c]: Yep you're right.

**Pull request merged**